### DISCUSSION TITLE
Best way to start a container with pre-configured realm

### DISCUSSION THREAD

**Discussion by benzen** [dasniko/testcontainers-keycloak#91] [Q&A]
I'd like to use this container to run my tests against keycloak.

For them to run I need to create a client with some config and an admin user.

I was hoping to export a configured realm and run it with the `withRealmImportFile`.

At first I wasn't recreating the realm because I used master as my realm and with this container master realm is already created, and keycloak doesn't allow to override a realm while importing.

Then I try to create the same realm with a different name. But now the user admin is created, but it's located in the master realm. And I noticed that my user even if they are in my realm.json, they are not imported.

What would be the recommended way to proceed ?
Should a start the container empty and build up the content of KC instance programatically ?

**Comment by dasniko** [comment]
I don't have issues using an exported realm json with complete configuration with clients and users importing it on startup with `.withRealmImportFile()`. This should be the way to go.
Never use the `master` realm for functional configurations, it's the management realm for Keycloak and should only be used for technical and administrational reasons.
